ui/app/src/components/EditGroupDialog/EditGroupDialogContainer.tsx

```diff
@@ -95,7 +99,11 @@ export function EditGroupDialogContainer(props: EditGroupDialogContainerProps) {
   }, [group?.id, props.group, setGroup]);
 
   useEffect(() => {
-    setSubmitOk(Boolean(group.name.trim() && !validateGroupNameMinLength(group.name)));
+    // This validation is different than groupName error, because for groupNameError it will return true only when form
+    // is dirty. For submit, it will be true even though form is not dirty (to avoid submitting a clean form).
+    setSubmitOk(
+      Boolean(group.name.trim() && !validateGroupNameMinLength(group.name) && !isInvalidGroupName(group.name))
+    );
```

Comment:
Doesn't make much sense to have an effect for this, might as well have it directly as a value like the `groupNameError` on the body of the component